source/com.microsoft.tfs.client.common/src/com/microsoft/tfs/client/common/credentials/CredentialsHelper.java

```diff
@@ -51,115 +47,53 @@
+        final CachedCredentials cachedCredentials = gitCredentialsManager.getCredentials(baseURI);
 
-            if (cachedCredentials.isPatCredentials()) {
-                gitCredentialsManager.removeCredentials(cachedCredentials);
-                createAccountCodeAccessToken(connection);
-            }
+        if (cachedCredentials == null) {
+            // No credentials are cached for Git.
+            // Let's use the current ones. They might be PAT, Alternative (on
+            // hosted), or Basic (on prem.)
+            gitCredentialsManager.setCredentials(currentCredentials);
         }
 
-    }
-
-    public static boolean hasAccountCodeAccessToken(final TFSConnection connection) {
-        if (connection.isHosted()) {
-            final URI baseURI = connection.getBaseURI();
-            final CachedCredentials cachedCredentials = gitCredentialsManager.getCredentials(baseURI);
-
-            if (cachedCredentials != null
-                && cachedCredentials.isPatCredentials()
-                && !StringUtil.isNullOrEmpty(cachedCredentials.getPassword())) {
-
-                return true;
-            }
+        if (cachedCredentials.equals(currentCredentials)) {
+            // The credentials haven't changed.
+            // No need to refresh.
+            return;
         }
 
-        return false;
-    }
-
-    public static boolean hasAlternateCredentials(final TFSConnection connection) {
-        if (connection.isHosted()) {
-            final URI baseURI = connection.getBaseURI();
-            final CachedCredentials cachedCredentials = gitCredentialsManager.getCredentials(baseURI);
-
-            if (cachedCredentials != null && cachedCredentials.isUsernamePasswordCredentials()) {
-                return true;
-            }
-        }
-
-        return false;
-    }
-
-    public static boolean isCredentialsValid(final TFSConnection connection, final Credentials credentials) {
-        final URI baseURI = connection.getBaseURI();
-        return isCredentialsValid(baseURI, credentials);
-    }
-
-    public static boolean isAccountCodeAccessTokenValid(final TFSConnection connection) {
-        if (connection.isHosted()) {
-            final URI baseURI = connection.getBaseURI();
-            final CachedCredentials cachedCredentials = gitCredentialsManager.getCredentials(baseURI);
-
-            if (cachedCredentials != null && cachedCredentials.isPatCredentials()) {
-                final PatCredentials patCredentials = (PatCredentials) cachedCredentials.toCredentials();
-                return isCredentialsValid(connection, PreemptiveUsernamePasswordCredentials.newFrom(patCredentials));
-            }
+        if (cachedCredentials.isPatCredentials() == currentCredentials.isPatCredentials()) {
+            // The credentials are changed and are of the same type, i.e either
+            // both PAT, or both Alternative, or both Basic. Let's refresh the
```

Comment:
the comment is misleading. It implies there are 3 types of creds (PAT, Alternate, Basic), but the if statement implies only 2 (PAT and not PAT). Perhaps the comment can be changed to make it clear?